blog-posts/staticIpLambda/staticIpLambda.md

```diff
@@ -0,0 +1,193 @@
+
+All the resources will be deployed using the **Serverless Framework** and the **Typescript AWS CDK**. If you are not familiar with it, I recommend you check out this [Serverless Framework x CDK blog post](https://dev.to/kumo/serverless-framework-aws-cdk-1dnf) to learn how to conciliate the two. 🤝
+
+First, we need to create a VPC with a public subnet and a private subnet. The private subnet will host our lambda function and the public subnet will host our NAT Gateway. <br>
+
+```ts
+const vpc = new Vpc(stack, 'Vpc', {
+	vpcName: 'vpc',
+	natGateways: 1, // 👈 Automatically creates an Elastic IP
+	maxAzs: 1, // 👈 Use more if you need high availability
+	subnetConfiguration: [
+		{
+		name: 'private-subnet-1',
+		subnetType: SubnetType.PRIVATE_WITH_NAT,
+		cidrMask: 24,
+		},
+		{
+		name: 'public-subnet-1',
+		subnetType: SubnetType.PUBLIC,
+		cidrMask: 24,
+		},
+	],
+});
+```
+> 💸 **Pricing Disclaimer**
+> This architecture will cost **some money**. The Nat Gateway runs on a EC2 instance which will cost you **around 30$/month**. This price is multiplied by the number of AZs you use. 
+ 
+ I also recommend that you use CloudFormation Outputs to store the IDs of the resources you will need later on. We want to slap that *✨clean IaC code✨* in our PR 🫱.
+ 
+ ```ts
+
+export const vpcSecurityGroupOutputId = 'SgOutputId';
+export const vpcPrivateSubnetOutputId = 'VpcPvSubOutputId';
+...
+
+const privateSubnets = vpc.selectSubnets(
+	{ subnetType: SubnetType.PRIVATE_WITH_NAT }
+);
+const [privateSubnetId1] = privateSubnets.subnetIds;
+
+new CfnOutput(stack, vpcSecurityGroupOutputId, {
+	value: vpc.vpcDefaultSecurityGroup,
+});
+new CfnOutput(stack, vpcPrivateSubnet1OutputId, {
+	value: privateSubnetId1,
+});
+
+```
+
+## ⚡ Deploy the Lambda function
+
+Now that we have our VPC, we can deploy our lambda function, using the Serverless Framework. 
+
+```ts
+
+// A small util to get CF template outputs 👇
+const getOutputValue = (
+	template: CloudFormationTemplate,
+	outputId: string
+ ): string => template.Outputs?.[outputId];
+
+export const staticIpLambda = {
+  timeout: 15,
+  handler: getHandlerPath(__dirname),
+  vpc: {
+    securityGroupIds: [
+		getOutputValue(
+		 cdkAppCloudFormationTemplate,
+		 vpcSecurityGroupOutputId
+		 ),
+	],
+    subnetIds: [
+		getOutputValue(
+		  cdkAppCloudFormationTemplate,
+		  vpcPrivateSubnet1OutputId
+		  ),
+		],
+  	},
+};
+```
+
+And bam! 🎉 Our lambda function is deployed in our private subnet. <br>
+All the outbound traffic from our lambda function will now go through the NAT Gateway, which will use the Elastic IP we created earlier. 🚀
+
+---
+
+You can retrieve the Elastic IP of your NAT Gateway using the AWS Console.
+Go to the VPC service and click Elastic IPs. You should see the Elastic IP created by the CDK. <br>
+
+![Elastic IP](./assets/elasticIp.png 'Elastic IP')
+
+Now just message your partner and ask them to whitelist this IP. 📩
+
+# 2️⃣ Perform SFTP operations in your Lambda function
```

Comment:
Bizarre la deuxième partie. Je me serais arrêté à la première. C'est plus facile à lire les articles court.

Le deuxième article tu peux faire un très court "how to put object in a sftp using Lambda". Tu rajoutes des liens entre les deux et ca te fait 2 articles pour le prix d'un